### PR TITLE
VIH-7409 - scheduler DI

### PR DESCRIPTION
**Pull request by levibutler**
### JIRA link (if applicable) ###
VIH-7409

### Change description ###
Scheduler DI change over, test cleanup, config changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[X] No
```

### CODE REVIEW

File: SchedulerJobs/SchedulerJobs.UnitTests/Functions/RunAnonymiseHearingsConferencesAndDeleteAadUsersFunctionTests.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+namespace SchedulerJobs.UnitTests.Functions
+{
+
+    public class RunAnonymiseHearingsConferencesAndDeleteAadUsersFunctionTests : AzureFunctionTestBaseSetup<AnonymiseHearingsConferencesAndDeleteAadUsersFunction>
+    {
+        [Test]
+        public async Task Timer_should_log_message_all_older_conferences_were_updated()
+        {
+            // Arrange
+            var usernames = new UserWithClosedConferencesResponse();
```
Comment (levibutler) [inline SchedulerJobs/SchedulerJobs.UnitTests/Functions/RunAnonymiseHearingsConferencesAndDeleteAadUsersFunctionTests.cs:20]: not needed

**Review by shaed-parkar — APPROVED** [review]
Nice work! Glad to see the tests for a bit of house keeping too!

File: SchedulerJobs/SchedulerJobs.UnitTests/Functions/RunCloseExpiredFunctionTests.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+namespace SchedulerJobs.UnitTests.Functions
+{
+    public class RunCloseExpiredFunctionTests : AzureFunctionTestBaseSetup<ClearHearingsFunction>
+    {
+        [Test]
+        public async Task Timer_should_log_message()
+        {
+            // Arrange
+            var result = new List<ExpiredConferencesResponse>();
+            _mocker.Mock<IVideoApiClient>().Setup(x => x.GetExpiredOpenConferencesAsync()).ReturnsAsync(result);
```
Comment (levibutler) [inline SchedulerJobs/SchedulerJobs.UnitTests/Functions/RunCloseExpiredFunctionTests.cs:20]: not needed and add verify

File: SchedulerJobs/SchedulerJobs.UnitTests/Functions/RunAnonymiseHearingsConferencesAndDeleteAadUsersFunctionTests.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                "username2@email.com",
+                "username3@email.com"
+            };
+            _mocker.Mock<IBookingsApiClient>().Setup(x => x.GetPersonByClosedHearingsAsync()).ReturnsAsync(usernames);
+
+            // Act
+            await _sut.Run(_timerInfo, _logger);
+
+            // Assert
+            _logger.GetLoggedMessages().Last().Should().StartWith("Data anonymised for hearings, conferences older than 3 months.");
```
Comment (levibutler) [inline SchedulerJobs/SchedulerJobs.UnitTests/Functions/RunAnonymiseHearingsConferencesAndDeleteAadUsersFunctionTests.cs:33]: add verify

**Review by levibutler — COMMENTED** [review]
LGTM

**Review by shaed-parkar — APPROVED** [review]

**Pull request merged**